addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -1759,3 +1759,34 @@ def test_multi_users_enabled(self):
         self.assertEqual(res_pick.move_line_ids, expected_backorder_mls)
         # Original picking in state done
         self.assertEqual(pick.state, "done")
+
+
+class TestStockPickingPriorities(common.BaseUDES):
+    def test_priority_remains_on_done_picking(self):
+        """
+        Test that priorities are not overwritten on done pickings.
+        """
+        self.create_quant(self.fig.id, self.test_stock_location_02.id, 100)
+        pick = self.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{"product": self.fig, "uom_qty": 10}],
+            location_id=self.test_stock_location_02.id,
+            priority="1",
+        )
+        pick.move_lines[0].quantity_done = 10
+        pick._action_done()
```

Comment:
Q: Is the picking in a done state after calling this? Doesn't it need to be confirmed and assigned stock first?

Reply:
Through my testing, this is enough to trigger the code that would reset the priority (prior to the fix)

Reply:
(So without the fix, this UT would fail as the priority would be reset)

Reply:
Can you verify that the picking is in a done state? Can either assert it or just debug locally. We should keep our tests as realistic as possible.

Reply:
I can confirm it puts it into state 'done'

Reply:
Can you confirm what the state was before calling `_action_done`?

Just not sure how the pick has move lines unless it has already been assigned stock, but from the earlier lines I can't see where this is being done.

Reply:
State draft.

Reply:
Ah I think I see another problem, you are setting the quantity done on the `stock.move` records, not `stock.move.line` records.

`move_lines` are moves  and `move_line_ids` are move lines, bad naming from Odoo but it is what it is.

So let's do this, make the picking confirmed and assigned in `create_picking`.

Remove the code where you are setting the quantity done and calling `_action_don`, and instead call `complete_picking`, this helper method will set the quantity done of all move lines properly and then call action done if validate is true (the default).

Reply:
Pushed changes. Not sure I am confirming and assigning before setting priority, as they are being passed in the same call to create_picking, just in that order... is that what you meant?

Reply:
Yeah I just meant the ordering, as priority is part of `kwargs` whereas `confirm` and `assign` are args specified in the method.

But looks good, thanks. :)